single_tech_samples/databricks/sample4_ci_cd/README.md

```diff
@@ -1 +1,256 @@
+# Azure Databricks CI/CD template
+
+## Contents
+
+- [1. Solution Overview](#1-solution-overview)
+  - [1.1. Scope](#11-scope)
+  - [1.2. Architecture](#12-architecture)
+  - [1.3. Technologies used](#13-technologies-used)
+- [2. How to use this template](#2-how-to-use-this-sample)
+  - [2.1. Prerequisites](#21-prerequisites)
+    - [2.1.1 Software Prerequisites](#211-software-prerequisites)
+  - [2.2. Setup and deployment](#22-setup-and-deployment)
+  - [2.3. Deployed Resources](#23-deployed-resources)
+  - [2.4. Sample project Structure](#24-sample-project-structure)
+    - [2.4.1 multiple notebooks](#241-multiple-notebooks)
+    - [2.4.2 notebook by pyspark API](#242-notebook-by-pyspark-API)
+    - [2.4.3 notebook by sparksql](#243-notebook-by-sparksql)
+    - [2.4.4 notebook plus python module](#244-notebook-plus-python-module)
+    - [2.4.5 python spark jo](#245-python-spark-job)
+
+## 1. Solution Overview
+
+When build a project in databricks, we can start from some notebooks and implement the business logic in python. Before go-production, we need to create CI/CD pipelines. To reduce the effort of build CI/CD pipelines, we build this git repository as template including sample noteboks and unit testing plus Azure DevOps yaml files as CI/CD pipelines.
+
+**It is the scaffolding of Azure databricks project.**
+
+To make it easy extendable, the notebooks and python code only contain super simple logic, and the unit testing is implemented by pytest and [nutter](https://github.com/microsoft/nutter)
+
+This template focuses on automating provisioning, CI/CD pipeline, and various approaches of databricks implementation.
+
+### 1.1. Scope
+
+The following list captures the scope of this template:
+
+1. Sample code
+    1. Notebook with pyspark API.
+    2. Notebook with spark sql.
+    3. Multi-NotBooks (main notebook plus reusable notebooks)
+    4. notebooks with python module
+    5. Non-Notebook, purely python
+2. Unit testing
+    1. pytest for python module
+    2. [nutter](https://github.com/microsoft/nutter) for notebooks
+3. Provision Azure Databricks environments of development, testing and production via provision pipepline.
+4. CI/CD pipelien to build and run testing automatically.
+
+Details about [how to use this sample](#3-how-to-use-this-sample) can be found in the later sections of this document.
+
+### 1.2. Architecture
+
+The below diagram illustrates the deployment process flow followed in this template:
+
+  ![architecture](images/architecture.png "architecture")
+
+### 1.3. Technologies used
+
+The following technologies are used to build this template:
+
+- [Azure Databricks](https://azure.microsoft.com/en-au/free/databricks/)
+- [Azure Resource Manager](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/overview)
+- [nutter](https://github.com/microsoft/nutter)
+
+## 2. How to use this template
+
+This section holds the information about usage instructions of this template.
+
+### 2.1. Prerequisites
```

Comment:
Should we mention Sample 1 and Sample 3 as the prereq? I can add a link in Sample 3 to Sample 4 as 'Next Steps' to cross reference

Reply:
We can reference it in the readme of provisioning